dbt/adapters/base/impl.py

```diff
@@ -1743,6 +1746,15 @@ def capabilities(cls) -> CapabilityDict:
     def supports(cls, capability: Capability) -> bool:
         return bool(cls.capabilities()[capability])
 
+    @available
+    @classmethod
+    def get_adapter_run_info(cls, config: RelationConfig) -> AdapterTrackingRelationInfo:
```

Comment:
`RelationConfig` doesn't feel like the correct type. I would have guessed this is some version of `NodeConfig`.

Reply:
I don't have a good handle on the differences here. When I did introspection, we are passing in a `RelationConfig`. I bet I'm missing some appreciation of all these different Config objects and perhaps I was "lead on" by the suggestion for `RelationConfig` in the original ticket. If you feel strongly on this type, please do offer your thoughts here!

Reply:
Since this will only be called in the context of a model it should always conform to `RelationConfig`